### DISCUSSION TITLE
Autocontrast for light variant

### DISCUSSION THREAD

**Discussion by robert-j-webb** [mantinedev/mantine#5944] [Feature requests]
Hi,

I noticed the `light` variant of my buttons is often inaccessible. Would it be nice if `autocontrast` worked with the `light` variant the same as the `filled` variant? 

See code: [default-variant-colors-resolver.ts](https://github.com/mantinedev/mantine/blob/e3e3bb834de1f2f75a27dbc757dc0a2fc6a6cba8/packages/%40mantine/core/src/core/MantineProvider/color-functions/default-variant-colors-resolver/default-variant-colors-resolver.ts#L71)

Just taking a cursory glance but doing some similar check to what is done for the `filled` variant like

```
const textColor = _autoContrast
  ? // Insert some calculation comparing background to text color
    ? 'var(--mantine-color-black)'
    : 'var(--mantine-color-white)'
  : 'var(--mantine-color-white)';
```

This would be a pretty positive change I think and not too much effort. 

What does the community think?

**Comment by rtivital** [comment]
It is not possible:
- light variant uses background color with transparency, actual color depends on the element where the component is used
- text color is the value that user specified via `color` prop, it is not black or white

**Comment by robert-j-webb** [comment]
Thanks for checking it out! I agree
